### DISCUSSION TITLE
How to theme-override a secondary button of type "primary"

### DISCUSSION THREAD

**Discussion by paul-gaspar** [tusen-ai/naive-ui#3299] [Q&A] [answered]
Hello!

I have a secondary button of type primary:

`<n-button secondary type="primary">`

How can I override the style in a NConfigProvider?

For the normal button I can use the `Button` key:

`Button: { textColor: red }`

But which key do I have to use in my case? `ButtonSecondaryTypePrimary: { textColor: red }` does not work.

Thank you!
Paul

**Comment by 07akioni** [comment, marked as answer]
Color of secondary button is derived by `Button.colorPrimary` automatically. It's not possible to change it without influence other buttons.

**Reply by 07akioni** [reply]
The reason is the count of combination of button's props are too much. It's impossible to provide every param for every combination, nor the theme variable count for button would reach 1000  easily.
